core/src/main/java/io/questdb/cutlass/text/CairoTextWriter.java

```diff
@@ -457,7 +457,8 @@ TableStructureAdapter of(ObjList<CharSequence> names, ObjList<TypeAdapter> types
 
             if (timestampIndex > -1) {
                 final TypeAdapter timestampAdapter = types.getQuick(timestampIndex);
-                if ((timestampAdapter.getType() != ColumnType.LONG && timestampAdapter.getType() != ColumnType.TIMESTAMP) || timestampAdapter == BadTimestampAdapter.INSTANCE) {
+                final int typeTag = ColumnType.tagOf(timestampAdapter.getType());
```

Comment:
tagOf excess

Reply:
it's justifiable to convert it to tag once and compare 3 times